src/Umbraco.Web.BackOffice/Controllers/UsersController.cs

```diff
@@ -147,30 +172,24 @@ internal static async Task<HttpResponseMessage> PostSetAvatarInternal(HttpReques
                 //generate a path of known data, we don't want this path to be guessable
                 user.Avatar = "UserAvatars/" + (user.Id + safeFileName).GenerateHash<SHA1>() + "." + ext;
 
-                using (var fs = System.IO.File.OpenRead(file.LocalFileName))
+                using (var fs = file.OpenReadStream())
                 {
                     mediaFileSystem.AddFile(user.Avatar, fs, true);
                 }
 
                 userService.Save(user);
-
-                //track the temp file so the cleanup filter removes it
-                tempFiles.UploadedFiles.Add(new ContentPropertyFile
```

Comment:
Was this not needed then?

Reply:
It is not needed anymore. The reason is because in framework all files were uploaded to a temp location before the controller action was executed. Therefore we had to clean up (using the `FileUploadCleanupFilterAttribute`).

This is not the case anymore, and we handle the stream directly in the action, so nothing to clean up,